lisa/target.py

```diff
@@ -211,61 +210,112 @@ class TestEnv(Loggable, HideExekallID):
+        # That res_dir is for the exclusive use of Target itself, it must not
+        # be used by users of Target
 
-        self.target_conf = target_conf
-        logger.info('Target configuration:\n%s', self.target_conf)
+        res_dir = res_dir or ArtifactPathManager().get_dir(
+            name='{}-{}'.format(type(self).__qualname__, name),
+            symlink=False,
+        )
+
+        self._res_dir = res_dir
+        os.makedirs(self._res_dir, exist_ok=True)
+        if os.listdir(self._res_dir):
+            raise ValueError('res_dir must be empty: {}'.format(self._res_dir))
 
         if plat_info is None:
             plat_info = PlatformInfo()
         else:
             # Make a copy of the PlatformInfo so we don't modify the original
             # one we were passed when adding the target source to it
             plat_info = copy.copy(plat_info)
+            logger.info('User-defined platform information:\n%s', plat_info)
+
         self.plat_info = plat_info
 
         # Take the board name from the target configuration so it becomes
         # available for later inspection. That board name is mostly free form
         # and no specific value should be expected for a given kind of board
         # (i.e. a Juno board might be named "foo-bar-juno-on-my-desk")
-        if board_name:
-            self.plat_info.add_src('target-conf', dict(name=board_name))
+        if name:
+            self.plat_info.add_src('target-conf', dict(name=name))
 
-        logger.info('User-defined platform information:\n%s', self.plat_info)
 
         self._installed_tools = set()
-        self.target = self._init_target(self.target_conf, self._res_dir)
+        self.target = self._init_target(
+                kind=kind,
+                name=name,
+                workdir=workdir,
+                device=device,
+                host=host,
+                port=port,
+                username=username,
+                password=password,
+                keyfile=keyfile,
+                devlib_platform=devlib_platform,
+                devlib_excluded_modules=devlib_excluded_modules,
+            )
 
         # Initialize binary tools to deploy
-        tools = self.target_conf.get('tools', [])
         if tools:
             logger.info('Tools to install: %s', tools)
             self.install_tools(target, tools)
 
-        # Autodetect information from the target, after the TestEnv is
+        # Autodetect information from the target, after the Target is
         # initialized. Expensive computations are deferred so they will only be
         # computed when actually needed.
-        self.plat_info.add_target_src(self, fallback=True)
+
+        rta_calib_res_dir = os.path.join(self._res_dir, 'rta_calib')
+        os.makedirs(rta_calib_res_dir)
+        self.plat_info.add_target_src(self, rta_calib_res_dir, fallback=True)
 
         logger.info('Effective platform information:\n%s', self.plat_info)
 
+    def __getattr__(self, attr):
```

Comment:
I remember us having a conversation about this but I can't remember the details. Could you remind me why we do this instead of inheriting from `devlib.target:Target`?